src/modules/vtol_att_control/vtol_att_control_main.cpp

```diff
@@ -531,6 +534,10 @@ void VtolAttitudeControl::task_main()
 			// update transition state if got any new data
 			if (got_new_data) {
 				_vtol_type->update_transition_state();
+
+				_vtol_type->process_mc_data();
+				fill_mc_att_rates_sp();
+				// TODO(simon@uaventure.com): Do we also need to update the fw setpoints?
```

Comment:
This is quite confusing. In my opinion, there should only be one "process_data" and one "fill_att_rates_sp" which should be called in both cases. The setpoints/controls should then be applied using the mc_att_ctl_weight.